### PR TITLE
Fixing shaders for lighting and inventory

### PR DESCRIPTION
**Pull request by richardba**
This fixes the z-fighting issue in the inventory, but fixes directional lighting as well

### CODE REVIEW

File: TombEngine/Shaders/Math.hlsli [hunk 1]
```diff
@@ -29,4 +29,106 @@ float LinearizeDepth(float depth, float nearPlane, float farPlane)
 	return (2.0f * nearPlane) / (farPlane + nearPlane - depth * (farPlane - nearPlane));
 }
 
+float3 mod289(float3 x) {
```
Comment (Kubsy) [inline TombEngine/Shaders/Math.hlsli:32]: I see these 2 funcs are named "mod289" what does it mean? I am not sure if the shaders have to be readable :P.

File: TombEngine/Shaders/Math.hlsli [hunk 2]
```diff
@@ -29,4 +29,106 @@ float LinearizeDepth(float depth, float nearPlane, float farPlane)
...
+  return x - floor(x * (1.0 / 289.0)) * 289.0;
+}
+
+float4 permute(float4 x) {
+     return mod289(((x*34.0)+10.0)*x);
+}
+
+float4 taylorInvSqrt(float4 r)
+{
+  return 1.79284291400159 - 0.85373472095314 * r;
```
Comment (Kubsy) [inline TombEngine/Shaders/Math.hlsli:46]: are these numbers constants of some sort? Is it not possible to round them up?

**Review by Kubsy — APPROVED** [review]

File: TombEngine/Shaders/Math.hlsli (hunk 1 above)
Comment (richardba) [reply TombEngine/Shaders/Math.hlsli:32]: mod289 is function from The Book of Shaders, it is a permutation on the scale of 289 units, per 289 times

**Comment by Sezzary** [comment]
Great! Just some usual formatting notes from me:
- Use tabs, not spaces.
- Keep braces on their own line.
- Function names should be in PascalCase.
- Avoid single-character variable names as they have little meaning.

**Comment by richardba** [comment]
>     * Use tabs, not spaces.
>     * Keep braces on their own line.
>     * Function names should be in PascalCase.

All done, I have explained the reasoning behind single char variable in Discord chat. Thanks for the review

**Pull request merged**